lib/tests/2wp.js

```diff
@@ -819,6 +821,37 @@ const execute = (description, getRskHost) => {
 
     });
 
+    it('should reject and not refund a pegout when a contract is trying to execute it', async () => {
+
+      // Arrange
+
+      const initial2wpBalances = await get2wpBalances(rskTxHelper, btcTxHelper);
+      const pegoutValueInSatoshis = MINIMUM_PEGOUT_AMOUNT_IN_SATOSHIS;
+
+      const { callReleaseBtcContract, creatorAddress } = await deployCallReleaseBtcContract(rskTxHelper);
+      const initialRskSenderBalanceInWeisBN = await rskTxHelper.getBalance(creatorAddress);
+
+      // Act
+
+      const callBridgeReleaseBtcMethod = callReleaseBtcContract.methods.callBridgeReleaseBtc();
+      const contractCallTxReceipt = await sendTransaction(rskTxHelper, callBridgeReleaseBtcMethod, creatorAddress, satoshisToWeis(pegoutValueInSatoshis));
+
+      // Assert
+
+      const contractAddressChecksummed = rskTxHelper.getClient().utils.toChecksumAddress(ensure0x(callReleaseBtcContract.options.address));
+      const expectedEvent = createExpectedReleaseRequestRejectedEvent(contractAddressChecksummed, pegoutValueInSatoshis, PEGOUT_REJECTION_REASONS.CALLER_CONTRACT);
+      const actualReleaseRequestRejectedEvent = getReleaseRequestRejectedEventFromContractCallTxReceipt(contractCallTxReceipt);
+      expect(actualReleaseRequestRejectedEvent).to.be.deep.equal(expectedEvent);
+
+      await assert2wpBalancesAfterPegoutFromContract(initial2wpBalances, pegoutValueInSatoshis);
+
+      // The rsk sender should lose the funds since there's no refund when a smart contract is trying to do a pegout
```

Comment:
Should we also be checking the contract address balance?

Reply:
Updated.